libraries/Microsoft.Bot.Builder.Ai.LUIS/LuisRecognizer.cs

```diff
@@ -66,26 +53,18 @@ public Task<(RecognizerResult recognizerResult, LuisResult luisResult)> CallAndR
             return Recognize(luisRequest, ct, _luisRecognizerOptions.Verbose);
         }
 
-        /// <summary>
-        /// Runs an utterance through a LUIS recognizer and returns the recognizer results.
-        /// </summary>
-        /// <param name="request">A LUIS request for an utterance.</param>
-        /// <param name="ct">A cancellation token.</param>
-        /// <param name="verbose">Whether to add metadata to the recognizer's results.</param>
-        /// <returns>The recognizer results and LUIS result.</returns>
-        private async Task<(RecognizerResult recognizerResult, LuisResult luisResult)> Recognize(LuisRequest request, CancellationToken ct, bool verbose)
+        private async Task<RecognizerResult> Recognize(LuisRequest request, CancellationToken ct, bool verbose)
         {
             var luisResult = await _luisService.QueryAsync(request, ct).ConfigureAwait(false);
-
             var recognizerResult = new RecognizerResult
             {
                 Text = request.Query,
-                AlteredText = luisResult.AlteredQuery,
+                AlteredText = luisResult.AlteredQuery ?? request.Query,
                 Intents = GetIntents(luisResult),
-                Entities = GetEntitiesAndMetadata(luisResult.Entities, luisResult.CompositeEntities, verbose)
+                Entities = GetEntitiesAndMetadata(luisResult.Entities, luisResult.CompositeEntities, verbose),
             };
-
-            return (recognizerResult, luisResult);
+            recognizerResult.Properties.Add("luisResult", luisResult);
+            return recognizerResult;
         }
 
         private static JObject GetIntents(LuisResult luisResult)
```

Comment:
Consider renaming. GetIntents, to me when I read this, is a method that goes out to LUIS and makes another round-trip. The semantics of this are "ExtractIntents", which is what this method is doing.

Reply:
Done